pygeos/lib/geos_wrapper.pxd

```diff
@@ -0,0 +1,9 @@
+cdef extern from "geos_c.h":
+    ctypedef void *GEOSContextHandle_t
+    GEOSContextHandle_t GEOS_init_r()
+    ctypedef struct GEOSGeometry
+
+    const GEOSGeometry* GEOSGetGeometryN_r(GEOSContextHandle_t handle, const GEOSGeometry* g, int n)
```

Comment:
Is it an idea to use the `except NULL` here and with the `GEOSGeom_clone_r` in order to not worry about errors in the code?

For `GEOSGetNumGeometries_r`, it is `except -1`, for `GEOSisEmpty_r` it is `except 2`

Reply:
Yes, good catch.  Put differently, we use `except` here when we want Cython to automatically catch those and raise exceptions for us.  There may be cases where we want to catch those errors ourselves to raise a specific exception, but these ones seem fine to handle automatically.